ros/src/twist_controller/dbw_node.py

```diff
@@ -129,6 +135,30 @@ def publish(self, throttle, brake, steer):
         bcmd.pedal_cmd = brake
         self.brake_pub.publish(bcmd)
 
+    def plot_velocity(self):
+        now = rospy.get_time()
+        diff_time = now - self.previous
+        self.previous = now
+
+        self.velocity_plot.append(np.array(self.current_twist.linear.x))
+        i = len(self.velocity_plot) - 1
+        self.acceleration_plot.append((self.velocity_plot[i] - self.velocity_plot[i-1])/diff_time)
+        self.jerk_plot.append((self.acceleration_plot[i] - self.acceleration_plot[i-1])/diff_time)
+
+        if i % 2000  == 0:
```

Comment:
@davidpilop Yep looks good to me, but If I am correct when @subhash suggested 
> Though a live plot looks very cool, a simpler solution would be just to accumulate 2000-3000 sample velocities, acc etc and just popping them up as one plot

he meant that once you handled `2000-3000` calls (up to you, in your case you decided to go with `2000` seems like) you should just execute the logic within this if condition statement and exit. So in here instead of checking for every `2000` calls with `if i % 2000  == 0:`, you should probably check for the first `2000` calls **only**, so you should replace `if i % 2000  == 0:` to `if i == 2000:` and also after you show statistics in line 160 (https://github.com/subhash/CarND-Capstone/pull/14/files#diff-7fdad75c966f68bdd51946534578bb1fR160) with `plt.show()` call you should exit by executing next instruction`exit(0)`.

@subhash could you have a look?

Reply:
@volkodava @davidpilop The condition is fine for now, we can always quit ROS altogether once we get the plot. 

Good job on getting the plot, but something is not right about the values. 
![see](https://user-images.githubusercontent.com/87746/37287244-57433302-262a-11e8-9b67-0147108ff010.png)

We cannot be introducing acc or jerk at this level! Can you investigate. Otherwise, I can look at it tomorrow

Reply:
I think something of this sort is required:

```
        if i>0:
            self.acceleration_plot.append((self.velocity_plot[i] - self.velocity_plot[i-1])/diff_time)
        if i>1:
            self.jerk_plot.append((self.acceleration_plot[i-1] - self.acceleration_plot[i-2])/diff_time)

```

Still, we are not great on the jerk front ..

Reply:
![see](https://user-images.githubusercontent.com/87746/37321609-159f4600-269f-11e8-8937-81ffc1f0a9a2.png)

The main part of the run

Reply:
> Still, we are not great on the jerk front ..
@subhash could you try to run the test with uncommented https://github.com/subhash/CarND-Capstone/blob/master/ros/src/waypoint_updater/waypoint_updater.py#L78 and commented out https://github.com/subhash/CarND-Capstone/blob/master/ros/src/waypoint_updater/waypoint_updater.py#L79 to verify if it helps with jerk?